languages/go/internal/ffi/ffi.go

```diff
@@ -23,10 +23,13 @@ import (
 /*
 Reads a c string from polar core to a go string and frees the c string.
 */
-func readStr(cStr *C.char) string {
+func readStr(cStr *C.char) *string {
+	if cStr == nil {
+		return nil
+	}
 	goStr := C.GoString(cStr)
 	C.string_free(cStr)
-	return goStr
+	return &goStr
```

Comment:
why?

Reply:
Because there's the next message API which returns an optional message, and so sometimes returns `nil`. So we need this codepath to make that correctly map `Option<C.char>` to `Option<string>`